Lib/test/datetimetester.py

```diff
@@ -16,6 +16,7 @@
 import re
 import struct
 import unittest
+from unittest.mock import ANY
```

Comment:
We have a [dedicated class for this in `datetimetester.py`](https://github.com/python/cpython/blob/4ef0e2308957fa358dc0e4a617ee4794b5230a2d/Lib/test/datetimetester.py#L58) to avoid relying on the implementation details of `ANY`.

I think we can move it into `tests.support` if you are going to be adding tests like this across multiple modules.

Reply:
`unittest.mock.ANY` only accidentally has the property that it compares equal to things. Even if that were a guaranteed property (it *probably* won't change, but it's an implementation detail), it's still confusing to see people using the "wild card for arguments" singleton from `unittest.mock` for one of its incidental properties (comparing equal to things).

If someone tried to understand this test and looked it up in the documentation, they would be confused as to what it means. I think it is much clearer to use a dedicated class for this (particularly since we already have one in the `datetime` tests.

Reply:
That class was added 15 days ago. Perhaps the author was not aware of `mock.ANY`. There are explicit tests for this property of `mock.ANY`, it is not incidental.

It is not very difficult to add a duplicate in `test.support`, but I am not sure that it is worth.

Reply:
> There are explicit tests for this property of `mock.ANY`, it is not incidental.

I don't know about that, people test implementation details all the time.

> That class was added 15 days ago. Perhaps the author was not aware of `mock.ANY`.

Definitely they were. The original PR used `mock.ANY` and I asked for it to be replaced with a dedicated class. You can see [my comment on the original PR](https://github.com/python/cpython/pull/14726#discussion_r303197977).

If you look at the [documentation for `ANY`](https://docs.python.org/3/library/unittest.mock.html#any), it is clearly documented as a wild card to be used for making assertions about *some* calls to a function but not others. Nothing in the documentation says that it is implemented with a permissive `__eq__`, and in fact a permissive `__eq__` is not always sufficient, as we learned in #14700. The `ANY` class does have the property you want, but it's a dedicated singleton with a semantic meaning other than "this compares equal to anything".

For reasons of clarity, I think it makes sense to have dedicated singletons for each of the custom comparison behaviors, even if there are other objects that also have those properties for their own reasons.

Reply:
This change has been extracted into #14996. Please take a look.